frontend/src/metabase/browse/components/CollectionBreadcrumbsWithTooltip.tsx

```diff
@@ -0,0 +1,151 @@
+import classNames from "classnames";
+import type { FC } from "react";
+import { forwardRef, useLayoutEffect, useRef, useState } from "react";
+
+import { ResponsiveContainer } from "metabase/components/ResponsiveContainer/ResponsiveContainer";
+import { useAreAnyTruncated } from "metabase/hooks/use-is-truncated";
+import resizeObserver from "metabase/lib/resize-observer";
+import * as Urls from "metabase/lib/urls";
+import type { FlexProps } from "metabase/ui";
+import { FixedSizeIcon, Flex, Group, Text, Tooltip } from "metabase/ui";
+import type { Collection } from "metabase-types/api";
+
+import { getCollectionName } from "../utils";
+
+import {
+  Breadcrumb,
+  CollectionBreadcrumbsWrapper,
+} from "./CollectionBreadcrumbsWithTooltip.styled";
+import type { RefProp } from "./types";
+import { getBreadcrumbMaxWidths } from "./utils";
+
+const separatorCharacter = "/";
+
+export const CollectionBreadcrumbsWithTooltip = ({
+  collection,
+  containerName,
+}: {
+  collection: Collection;
+  containerName: string;
+}) => {
+  const collections = (collection.effective_ancestors || []).concat(collection);
+  const pathString = collections
+    .map(coll => getCollectionName(coll))
+    .join(` ${separatorCharacter} `);
+  const ellipsifyPath = collections.length > 2;
+  const shownCollections = ellipsifyPath
+    ? [collections[0], collections[collections.length - 1]]
+    : collections;
+  const justOneShown = shownCollections.length === 1;
+
+  const { areAnyTruncated, ref } = useAreAnyTruncated<HTMLAnchorElement>({
+    tolerance: 1,
+  });
+
+  const initialEllipsisRef = useRef<HTMLDivElement | null>(null);
+  const [
+    isFirstCollectionDisplayedAsEllipsis,
+    setIsFirstCollectionDisplayedAsEllipsis,
+  ] = useState(false);
+
+  useLayoutEffect(() => {
+    const initialEllipsis = initialEllipsisRef.current;
+    if (!initialEllipsis) {
+      return;
+    }
+    const handleResize = () => {
+      // The initial ellipsis might be hidden via CSS,
+      // so we need to check whether it is displayed via getComputedStyle
+      const style = window.getComputedStyle(initialEllipsis);
+      setIsFirstCollectionDisplayedAsEllipsis(style.display !== "none");
+    };
+    resizeObserver.subscribe(initialEllipsis, handleResize);
+    return () => {
+      resizeObserver.unsubscribe(initialEllipsis, handleResize);
+    };
+  }, [initialEllipsisRef]);
```

Comment:
This uses the same pattern as useIsTruncated. If you think it's a performance concern, I'd say we could ask Dawei if we could simply remove this feature. The feature is this: the first collection in the list truncates down to "..." when the cell is too small. I think it's relatively low in value and it would be a shame if performance was compromise for the sake of it. I'll leave it for now